colaraz/cms/static/js/header.js

```diff
@@ -116,3 +120,72 @@ function removeNotificationsCount() {
         markNotificationsAsRead();
     }
 }
+
+function getAndPopulateJobAlerts() {
+    $.ajax({
+        type: "GET",
+        url: colarazJobAlertsFetchingUrl,
+        success: function (resp) {
+            let job_alerts = "";
+            if (resp.d.length) {
+                resp.d.forEach(element => {
+                    job_alerts += createJobAlert(element.Image, element.Heading, element.Message, element.RelativeTime);
+                });
+            }else{
+                job_alerts = "<li><p>No new job alerts</p></li>"
+            }
+
+            handleJobAlertsListing(job_alerts);
+            handleJobAlertsCount(resp.d.length);
+
+        },
+        error: function (resp) {
+            console.error(`Job Alerts API gave following error: ${resp.Message}`);
+        },
+    });
+
+    function handleJobAlertsListing(job_alerts) {
+        $("#job-alerts-list").html(job_alerts);
+    }
+    function handleJobAlertsCount(count) {
+        if (count == 0) {
+            let countSpan = $("#unread-job-alerts-count");
+            countSpan.removeClass("count");
+            countSpan.html("");
+        } else {
+            let countSpan = $("#unread-job-alerts-count");
+            countSpan.addClass("count");
+            countSpan.html(count);
+        }
+    }
+
+    function createJobAlert(img_src, heading, message, relativeTime) {
+        return `<li>` +
+                    `<div class="media">` +
+                        `<a href="#">` +
+                            `<img src="${img_src ? img_src : "https://placehold.it/34x34"}" alt="">` +
+                        `</a>` +
+                    `</div>` +
+                    `<div style="font-size: 12px;padding: 0;color: #333;" class="description">` +
+                        `${heading} <span style="color: #aaaaaa;font-family: sans-serif;"> at </span> ${message}` +
+                        `<p><i class="fa fa-share-square-o"></i> ${relativeTime}</p>` +
+                    `</div>` +
+                `</li>`;
+    }
+}
+
+function markJobAlertsAsRead() {
+    $.ajax({
+        type: "GET",
+        url: colarazJobAlertsMarkingUrl
+    });
```

Comment:
Can we add logging here in cases of success and failures ?

Reply:
Logging has been done on server (edx-platform) side, I don't think its a good idea to shore error to users.

Reply:
There is no harm in it as edX does this too. and what if the request never reaches the server? logging here could help identify issues like that. just saying.

Reply:
This API call will be done after every 30 seconds, obviously, there is no harm but still, it will be shown to the user which I personally don't like because proper logging is being done on the server end.
As far as the scene where the request doesn't reach the server, I believe the browser itself gives error in the console about that, it doesn't need to be managed.
But as it may be a concern to most of the people, I am adding the log now.

Reply:
done.